src/components/store.js

```diff
@@ -0,0 +1,174 @@
+      'parking': 'https://patterns.boston.gov/images/global/icons/mapping/parking.svg',
+    },
+    dataSourceTypeChoices: {
+      'CoB ArcGIS': 'cob-arcgis'
+    },
+    polygonStyles: {
+      'default': {
+        uid: 'default',
+        color: '#0C2639',
+        hoverColor: '#FB4D42',
+      },
+    },
+    polygonStyleChoices: {
+      'Default Style': 'default',
+    },
+  },
+  getters: {
+    allDataSources: state => {
+      return state.dataSourceList.map(dataSourceId => state.dataSources[dataSourceId])
+    },
+    allMaps: state => {
+      return state.mapList.map(mapId => state.maps[mapId])
+    },
+    getDataSourceById: (state) => (uid) => {
+      return state.dataSources[uid]
+    },
+    getMapById: (state) => (uid) => {
+      return state.maps[uid]
+    },
+    getPolygonStyleById: (state) => (uid) => {
+      return state.polygonStyles[uid]
+    },
+    getConfig: (state, getters) => {
+      let config = {
+        vizId: state.vizId,
+        title: state.title,
+        description: state.description,
+        dataSources: [],
+        maps: [],
+      }
+      state.dataSourceList.forEach(uid => {
+        config.dataSources.push(getters.getDataSourceConfig(uid))
+      })
+      state.mapList.forEach(uid => {
+        config.maps.push(getters.getMapConfig(uid))
+      })
+      return config
+    },
+    getDataSourceConfig: (state) => (uid) => {
+      let datasource = state.dataSources[uid]
+      return {
+        uid: datasource.uid,
+        type: datasource.dataSourceType,
+        icon: datasource.icon,
+        clusterIcons: datasource.clusterPoints,
+        polygonStyle: state.polygonStyles[datasource.polygonStyle],
+        popover: datasource.popover,
+        attributes: Object.assign({}, datasource.attributes),
+        legendLabel: datasource.legendLabel,
+      }
+    },
+    getMapConfig: (state) => (uid) => {
+      let map = state.maps[uid]
+      return {
+        uid: map.uid,
+        latitude: 42.347316,
+        longitude: -71.065227,
+        zoom: 12,
+        showZoomControl: map.showZoomControl,
+        showLegend: map.showLegend,
+        findUserLocation: map.findUserLocation,
+        searchForAddress: map.searchForAddress,
+        zoomToAddress: map.zoomToAddress,
+        placeholderText: map.placeholderText,
+        showDataLayer: map.showDataLayer,
+      }
+    },
+  },
+  mutations: {
+    SET_VIZID: (state, payload) => {
+      if (payload) {
+        state.vizId = payload
+      } else {
+        state.vizId = nanoid()
+      }
+    },
+    SET_TITLE: (state, payload) => {
+      state.title = payload || ''
+    },
+    SET_DESCRIPTION: (state, payload) => {
+      state.description = payload || ''
+    },
+    UPDATE_DATASOURCE: (state, payload) => {
+      if (!payload.hasOwnProperty('uid')) {
+        payload.uid = nanoid()
+      }
+      state.dataSources = {...state.dataSources, [payload.uid]: payload}
```

Comment:
Does Vuex handle cloning `payload` in this case?

Reply:
Yes.